elastalert/elastalert.py

```diff
@@ -310,16 +325,20 @@ def remove_duplicate_events(self, data, rule):
 
         # Remember the new data's IDs
         for event in data:
-            rule['processed_hits'][event['_id']] = event['_source'][rule['timestamp_field']]
-
-        return [event['_source'] for event in data]
+            rule['processed_hits'][event['_id']] = event['fields'][rule['timestamp_field']]
+        return [event for event in data]
 
     def remove_old_events(self, rule):
         # Anything older than the buffer time we can forget
         now = ts_now()
         remove = []
         buffer_time = rule.get('buffer_time', self.buffer_time)
+        if rule.get('timestamp_type','datetime') == 'long':
+            now = dt_to_int(now)
+            buffer_time = timedelta_to_int(buffer_time)
         for _id, timestamp in rule['processed_hits'].iteritems():
+            if rule.get('timestamp_type','datetime') == 'long':
+                timestamp = dt_to_int(timestamp)
```

Comment:
Why convert now, buffer_time and timestamp to ints, when ``now - timestamp > buffer_time`` will still have the same results?